apps/gitness/src/pages/pipeline-create/PipelineCreateDialog/PipelineCreateForm.tsx

```diff
@@ -0,0 +1,171 @@
+import { useEffect } from 'react'
+import {
+  Button,
+  Checkbox,
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+  Input,
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+  useZodForm
+} from '@harnessio/canary'
+import { z } from 'zod'
+import { useListBranchesQuery } from '@harnessio/code-service-client'
+import { CreateFormType } from '../../../types'
+
+interface PipelineCreateFormProps {
+  onCancel: () => void
+  onSubmit: (formValues: CreateFormType) => Promise<void>
+}
+
+const createPipelineSchema = z.object({
+  name: z.string().min(1, { message: 'Pipeline name is required' }),
+  branch: z.string().min(1, { message: 'Branch name is required' }),
+  yamlPath: z.string().min(1, { message: 'YAML path is required' }),
+  storeInGit: z.boolean().optional(),
+  cloneRepo: z.boolean().optional()
+})
+
+export function PipelineCreateForm({ onCancel, onSubmit }: PipelineCreateFormProps) {
+  const form = useZodForm({
+    schema: createPipelineSchema,
+    defaultValues: {
+      name: '',
+      branch: 'main',
+      yamlPath: ''
+    }
+  })
+
+  const { data, isLoading } = useListBranchesQuery({ repo_ref: 'MyNewProject/MyRepo/+', queryParams: {} })
+
+  const { watch, setValue, clearErrors, trigger } = form
+
+  useEffect(() => {
+    const subscription = watch((value, { name }) => {
+      if (name === 'name') {
+        setValue('yamlPath', value.name ? `.harness/${value.name}.yaml` : '')
+
+        /**
+         * This is to validate YAML path field
+         */
+        trigger('yamlPath')
+      }
+    })
+    return () => subscription.unsubscribe()
+  }, [watch, setValue, clearErrors])
+
+  return (
+    <Form className="space-y-6" form={form} onSubmit={onSubmit}>
+      <div className="grid gap-2">
+        <FormField
+          control={form.control}
+          name="name"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel className="text-primary">Pipeline name*</FormLabel>
+              <FormControl>
+                <Input className="text-primary" {...field} />
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+
+        <FormField
+          control={form.control}
+          name="storeInGit"
```

Comment:
is `storeInGit` required for gitness? i doubt it.. all pipelines will be in git by default

Reply:
In the mock we had it. We can discuss and remove it